fontir/src/glyph.rs

```diff
@@ -596,7 +597,7 @@ fn apply_optional_transformations(
         }
     }
 
-    if context.flags.contains(Flags::FLATTEN_COMPONENTS) {
+    if context.flags.contains(Flags::FLATTEN_COMPONENTS) || source_args.flatten_components {
```

Comment:
Having multiple places to store the same flag is a pattern I've been burned by, and seen others be burned by, many times. How might we avoid this? Can we perhaps, maybe in a follow-on, get rid of context.flags and *only* have fields on static metadata?

Reply:
yea I think unifying this somehow would make a lot of sense.